1. The Smart Attendance Logger
    - The Goal: Manage a classroom attendance list and ensure no duplicate entries are recorded if a student taps their ID
card twice.
- Concepts: set , function , while.
- The Problem: Create a function mark_attendance() that keeps asking for student names until the user
types "STOP". Store names in a Set to automatically prevent duplicates.
- Input: Multiple strings (names) entered one by one.
- Output: The final count of unique students and the set itself.

In [ ]:
total_students = set()
def mark_attendance():
    while True:
        name = input("Enter the name :")
        if name.lower().strip() == "stop":
            break
        total_students.add(name)

mark_attendance()
mark_attendance()
mark_attendance()
print(f"Total Number of students : {len(total_students)}")
for i,name in enumerate(total_students, start = 1):
    print(f"{i}. {name.capitalize()}")        


Total Number of students : 4
1. Rintam
2. Baldeep
3. Vikas
4. Isha


2. The Restaurant Bill Splitter

    - The Goal: Calculate how much each person owes, including a tax and a custom tip percentage.

    - Concepts: Functions, default values, lists.

    - The Problem: Write a function split_bill(items_prices, people, tax=0.05, tip=0.10).

    - Input: A list like [25.0, 15.0, 10.0] and an integer for people.
    - Output: A single float (the amount per person).

In [6]:
def split_bill(items_price, people, tax = 0.05, tip = 0.10):
    total = sum(items_price)

    total_tax = total * tax
    total_tip = total * tip 
    
    total_bill = total + total_tax + total_tip 
    return total_bill/ people 


people = int(input("Enter total number of people : "))
items_price = []
while True:
    item = input("Enter items")
    if item.lower().strip() == "over":
        break
    items_price.append(int(item))

print(f"Total number of people : {people}")
print(f"Total items : {items_price}")
print(f"Each person will get : {split_bill(items_price=items_price, people= people)}")

Total number of people : 10
Total items : [100, 200, 300, 400, 50, 60, 13, 12, 143, 1234, 145, 155, 45, 65]
Each person will get : 336.03


### **3. The Digital Wardrobe Organizer**

**The Goal:**  
Check if a specific outfit is available or if parts of it are missing.

**Concepts:**  
Multiple `if` statements, `list`, `input()`.

**The Problem:**  
You have a list: `wardrobe = ["Blue Jeans", "White Shirt", "Black Jacket"]`.  
1. Ask the user for **3 items** they want to wear.  
2. Check each item individually against the list.  

**Input:**  
Three separate `input()` strings (e.g., "Red Tie", "White Shirt", "Socks").

**Output:**  
A "Missing" warning for every item not found in the list.

**Hint:**  
Use the `not in` operator to identify missing items.

In [10]:
wardrobe = ["Blue Jeans", "White Shirt", "Black Jacket"]

while True:
    outfit = input("Enter item (or 'over' to stop): ").strip()
    
    if outfit.lower() == "over":
        break 
    
    # Create a lowercase version of the list for comparison
    if outfit.lower() not in [item.lower() for item in wardrobe]:
        print(f"Warning !! Missing : {outfit}")

Warning !! Missing : blue
Warning !! Missing : iver


4. The E-Commerce Discount Engine
    - The Goal: Apply different discount rates based on the total purchase amount.
    - Concepts: if-elif-else , functions.
    - The Problem: Create a function apply_discount(total).
        - If total > $500: 20% off.,,,
        - If total > $200: 10% off...
        - Else: No discount.
    - Input: A single number (the total bill).
    - Output: The final price after the discount is subtracted.

In [14]:
def apply_discount(total):
    if total >= 500:
        total -= total * 0.20
    elif total < 500 and total >= 200:
        total -= total * 0.10
    else:
        return "No discount."
    
    return total

while True:
    total = int(input("Enter total amount: "))
    if total == 0:
        break 
    print(f"Total : {total}, After Discount : {apply_discount(total)}")



Total : 100, After Discount : No discount.
Total : 200, After Discount : 180.0
Total : 300, After Discount : 270.0
Total : 400, After Discount : 360.0
Total : 500, After Discount : 400.0
Total : 600, After Discount : 480.0
Total : 700, After Discount : 560.0


5. The Contact Book Search
    - The Goal: Look up a phone number using a person's name.
    - Concepts: dict , input , if-else.
    - The Problem: Create a Dictionary of 5 friends (Name: Number). Ask the user for a name to search.
    - Input: A string (Name).
    - Output: The phone number if found, or "Contact not found".

In [17]:
contacts = {
    "isha": 9571512780,
    "chahat": 6376780848,
    "baldeep": 9079867235,
    "harsh": 7357703637
}

contacts.get(input("Enter name : ").strip().lower(),"No contacts Found.")

9571512780

6. The Travel Itinerary Builder
    - The Goal: Print a numbered list of travel destinations for a user.
    - Concepts: for loop, list , enumerate.
    - The Problem: Given a list cities = ["Tokyo", "Paris", "New York", "London"]
    - Input: None (Uses the hardcoded list).
    - Output: A vertically formatted list (e.g., "1. Tokyo").

In [18]:
cities = ["Tokyo", "Paris", "New York", "London"]
for i, city in enumerate(cities, start = 1):
    print(f"{i}. {city}")

1. Tokyo
2. Paris
3. New York
4. London


7. The ATM Pin Validator
- The Goal: Lock the user out after 3 incorrect attempts.
- Concepts: while , if-else , break .
- The Problem: Set correct_pin ="1234" . Allow the user 3 tries to enter the right PIN.
- Input: Up to 3 string inputs.
- Output: "Welcome" on success, or "Card Blocked" after the 3rd fail.

In [2]:
correct_pin = "15150109"
attempts = 0
while True:
    if attempts == 3:
        print("Card Blocked !!")
        break

    pin = input("Enter pin : ").strip()
    if pin == correct_pin:
        print("Welcome, Your pin is correct !!")
    else:
        print(f"{pin} is Incorrect pin !!!")
    attempts += 1

1234 is Incorrect pin !!!
12346 is Incorrect pin !!!
12688 is Incorrect pin !!!
Card Blocked !!
